zulipterminal/ui_tools/messages.py

```diff
@@ -729,6 +730,36 @@ def main_view(self) -> List[Any]:
                 "/me", f"<strong>{self.message['sender_full_name']}</strong>", 1
             )
 
+        # If message contains submessages (like polls, todo), process them
+        # and update the message content.
+        if self.message["submessages"]:
```

Comment:
I missed this point.
A lot of test cases failed due to `KeyError` at this line. You can try using `get()` instead of using key to fetch the data from the dictionary